### PR TITLE
Sync: User meta

### PR DESCRIPTION
**Pull request by enejb**
Add user meta syncing to users.

- [x] Remove roles, capabilities and local from the user data
- [x] Add syncing of user meta on full sync.

#### Testing instructions:
* Do the test pass?

### CODE REVIEW

File: sync/class.jetpack-sync-defaults.php [hunk 1]
```diff
@@ -322,6 +322,25 @@ public static function get_post_meta_whitelist() {
 		return apply_filters( 'jetpack_sync_post_meta_whitelist', self::$post_meta_whitelist );
 	}
 
+	static $user_meta_whitelist = array(
+		'locale',
+		'*_user_level',
```
Comment (lezama) [inline sync/class.jetpack-sync-defaults.php]: Maybe we could whitelist `$wpdb->get_blog_prefix() . 'user_level'` and $wpdb->get_blog_prefix() . 'capabilities' ?

File: sync/class.jetpack-sync-module-users.php [hunk 2]
```diff
@@ -53,8 +53,32 @@ public function init_listeners( $callable ) {
...
+	public function is_whitelisted_user_meta( $meta_key ) {
+		$user_meta_keys = (array)Jetpack_Sync_Settings::get_setting( 'user_meta_whitelist' );
+		$user_keys = array_map( array( $this, 'map_user_key' ), $user_meta_keys );
+
+		return in_array( $meta_key, $user_keys );
+	}
+
+	public function map_user_key( $key ) {
+		global $wpdb;
+		return str_replace( '*_', $wpdb->get_blog_prefix(), $key );
```
Comment (lezama) [inline sync/class.jetpack-sync-module-users.php]: we could remove this method if we whitelist the full key

File: sync/class.jetpack-sync-settings.php [hunk 3]
```diff
@@ -19,6 +19,7 @@ class Jetpack_Sync_Settings {
 		'disable'                 => true,
 		'render_filtered_content' => true,
 		'post_meta_whitelist'     => true,
+		'user_meta_whitelist'     => true,
```
Comment (lezama) [inline sync/class.jetpack-sync-settings.php]: are sync settings per site on a multisite install?

**Comment by lezama** [comment]
This is looking great :)

File: sync/class.jetpack-sync-settings.php (hunk 3 above)
Comment (enejb) [reply sync/class.jetpack-sync-settings.php]: They are per site and not per network.

File: sync/class.jetpack-sync-defaults.php (hunk 1 above)
Comment (enejb) [reply sync/class.jetpack-sync-defaults.php]: I updated it so that when every-time we get the whitelist we return the values with the prefixed defined.

**Comment by oskosk** [comment]
@enejb this one needs a rebase !

**Comment by enejb** [comment]
Thanks @oskosk Lets hold off from merging this one just yet. 
we have some good ideas but it doesn't solve our issue. In terms of user capabilities that we want to see.

**Comment by lezama** [comment]
closing this in favour of https://github.com/Automattic/jetpack/pull/7937

**Pull request closed without merging**